### PR TITLE
Live reload font size from alacritty config

### PR DESCRIPTION
**Pull request by waynr**
Addresses first part of \#2737. Second part, which will be font family and styles to also be live-reloaded, is coming in a follow-up PR.

### CODE REVIEW

File: alacritty_terminal/src/term/mod.rs [hunk 1]
```diff
@@ -933,6 +933,11 @@ impl Term {
                 self.colors[i] = self.original_colors[i];
             }
         }
+        if self.original_font_size != config.font.size && self.original_font_size == self.font_size
```
Comment (chrisduerr) [inline alacritty_terminal/src/term/mod.rs]: It's probably not worth it to check for `self.original_font_size != config.font.size`, since all that saves us are two assignments which should basically be free. The font itself is only ever changed if it actually changed something anyways.

I'd also like to put this at the end of the method with one newline separating it. Just to separate the different blocks a bit from each other and not have one massive function without a single newline.

File: alacritty_terminal/src/term/mod.rs [hunk 2]
```diff
@@ -933,16 +933,17 @@ impl Term {
...
-            self.font_size = config.font.size;
-        }
         self.visual_bell.update_config(config);
         self.default_cursor_style = config.cursor.style;
         self.dynamic_title = config.dynamic_title();
         self.auto_scroll = config.scrolling.auto_scroll;
         self.grid.update_history(config.scrolling.history() as usize, &self.cursor.template);
+
+        if self.original_font_size == self.font_size
+        {
```
Comment (chrisduerr) [inline alacritty_terminal/src/term/mod.rs]: Alacritty's (and Rust's) codestyle is to open parenthesis on the same line. Running `cargo fmt` with `rustfmt` installed should fix this.

Comment (waynr) [reply alacritty_terminal/src/term/mod.rs]: Okay; my editor was running `rustfmt` automatically before I started working on this but it was also making a bunch of other trivial changes (and preferring the style shown here). I switched from `stable` to `nightly` and it seems to do what you want.

File: alacritty_terminal/src/selection.rs [hunk 3]
```diff
@@ -579,7 +579,7 @@ mod test {
         assert_eq!(selection.to_span(&term(5, 10)).unwrap(), Span {
             start: Point::new(0, Column(4)),
             end: Point::new(2, Column(4)),
-            is_block: true,
+            is_block: true
```
Comment (chrisduerr) [inline alacritty_terminal/src/selection.rs]: Rustfmt needs to be run with nightly to include the nightly formatting options. That's why it has removed the trailing semicolons here for you. Please undo these changes either manually or by running rustfmt with a nightly compiler.

Comment (waynr) [reply alacritty_terminal/src/selection.rs]: Hmm, I was running with nightly:

```
zsh/4 9472  (git)-[live-reload-font-size]-% rustup update nightly-x86_64-unknown-linux-gnu
info: syncing channel updates for 'nightly-x86_64-unknown-linux-gnu'

  nightly-x86_64-unknown-linux-gnu unchanged - rustc 1.40.0-nightly (702b45e40 2019-10-01)

info: checking for self-updates
zsh/4 9473  (git)-[live-reload-font-size]-% rustup which cargo
/home/wayne/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/bin/cargo
zsh/4 9474  (git)-[live-reload-font-size]-% /home/wayne/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/bin/cargo fmt
zsh/4 9475  (git)-[live-reload-font-size]-% git diff
zsh/4 9476  (git)-[live-reload-font-size]-%
```

> Please undo these changes either manually or by running rustfmt with a nightly compiler.

I'd be happy to remove them but I want to make sure we have a common understanding that I was using nightly. Also, it seems like the next person to run `cargo fmt` will produce the same changes anyway.

Comment (waynr) [reply alacritty_terminal/src/selection.rs]: Do you think it would be useful to add a travis check that fails if `rustfmt --check` returns non-zero?

Comment (kchibisov) [reply alacritty_terminal/src/selection.rs]: We have this check on CI, it's optional though, because we're running it on nightly I guess.

Comment (waynr) [reply alacritty_terminal/src/selection.rs]: Ah I didn't think to look at what was already running in travis. Looks like the `rustfmt` check is currently passing: https://travis-ci.org/jwilm/alacritty/jobs/592595414

Comment (chrisduerr) [reply alacritty_terminal/src/selection.rs]: Make sure to install rustfmt through `rustup` to make sure it's pulling the correct version.

File: alacritty_terminal/src/term/mod.rs [hunk 4]
```diff
@@ -938,6 +938,11 @@ impl Term {
         self.dynamic_title = config.dynamic_title();
         self.auto_scroll = config.scrolling.auto_scroll;
         self.grid.update_history(config.scrolling.history() as usize, &self.cursor.template);
+
+        if self.original_font_size == self.font_size {
+            self.original_font_size = config.font.size;
```
Comment (chrisduerr) [inline alacritty_terminal/src/term/mod.rs]: I found one issue with the functionality of this PR. I think the original font size should be changed regardless of the current font size. So whenever resetting to the original, it would reset to the one set in the config.

Comment (waynr) [reply alacritty_terminal/src/term/mod.rs]: Oh, good catch I'll address that shortly.

File: alacritty_terminal/src/selection.rs (hunk 3 above)
Comment (waynr) [reply alacritty_terminal/src/selection.rs]: That is how I have it installed. One possibility here is that I had an older version of `nightly` installed before this discussion and I updated it specifically to try to fix this; it may be that the older `nightly` was more opinionated about commas whereas the new nightly will allow either style. Whatever the case, I manually reverted the changes in a newer commit.

File: alacritty_terminal/src/term/mod.rs [hunk 5]
```diff
@@ -938,6 +938,11 @@ impl Term {
         self.dynamic_title = config.dynamic_title();
         self.auto_scroll = config.scrolling.auto_scroll;
         self.grid.update_history(config.scrolling.history() as usize, &self.cursor.template);
+
+        self.original_font_size = config.font.size;
+        if self.original_font_size == self.font_size {
```
Comment (chrisduerr) [inline alacritty_terminal/src/term/mod.rs]: We need to check first and set it later, not set it first and then check. Otherwise we'll compare to the changed value.

**Comment by chrisduerr** [comment]
Thanks for the PR, everything working great now. LGTM.
